client/src/components/ProductDetailPage.jsx

```diff
@@ -122,15 +129,15 @@ class ProductDetailPage extends React.Component {
   render() {
     return (
       <React.Fragment>
-        <Overview top={this} productInfo={productInfoMock} productStyles={productStylesMock} reviewsMeta={reviewsMetaMock}/>
-        <RelatedItemsAndOutfit
+        <OverviewWithTracking top={this} productInfo={productInfoMock} productStyles={productStylesMock} reviewsMeta={reviewsMetaMock}/>
+        <RelatedItemsAndOutfitWithTracking
           top={this}
           product={this.state.products[this.state.currentProductId]}
           relatedProducts={this.state.relatedProducts.map(productId => this.state.products[productId])}
           outfitProducts={this.state.yourOutfit.map(productId => this.state.products[productId])}
         />
-        <QuestionsAndAnswers questionsInfo={questionsMock} productInfo={productInfoMock}/>
-        <RatingsAndReviews onChangeFileHandler={this.onChangeFileHandler} onClickUploadHandler={this.onClickUploadHandler} productId={reviewsMock.product} reviews={reviewsMock.results} reviewsMeta={reviewsMetaMock}/>
+        <QuestionsAndAnswersWithTracking questionsInfo={questionsMock} productInfo={productInfoMock}/>
+        <RatingsAndReviewsWithTracking onChangeFileHandler={this.onChangeFileHandler} onClickUploadHandler={this.onClickUploadHandler} productId={reviewsMock.product} reviews={reviewsMock.results} reviewsMeta={reviewsMetaMock}/>
```

Comment:
That's really cool that it can just wrap the component with no change to the tests or other references in the app.